examples/stm32/nucleo-h743zi-baremetal/hal.h

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2022-2023 Cesanta Software Limited
+// All rights reserved
+//
+// Datasheet: RM0433, devboard manual: UM2407
+// https://www.st.com/resource/en/reference_manual/rm0433-stm32h742-stm32h743753-and-stm32h750-value-line-advanced-armbased-32bit-mcus-stmicroelectronics.pdf
+// Alternate functions: https://www.st.com/resource/en/datasheet/stm32h743vi.pdf
+
+#pragma once
+
+#include <stm32h743xx.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#define BIT(x) (1UL << (x))
+#define SETBITS(R, CLEARMASK, SETMASK) (R) = ((R) & ~(CLEARMASK)) | (SETMASK)
+#define PIN(bank, num) ((((bank) - 'A') << 8) | (num))
+#define PINNO(pin) (pin & 255)
+#define PINBANK(pin) (pin >> 8)
+
+// System clock (2.1, Figure 1; 8.5, Figure 45; 8.5.5, Figure 47; 8.5.6, Figure
+// 49) SYS_FREQUENCY <= 480 MHz; hclk = SYS_FREQUENCY / HPRE ; hclk <= 240 MHz;
+// APB clocks <= 120 MHz. D1 domain bus matrix (and so flash) runs at hclk
+// frequency. Configure flash latency (WS) in accordance to hclk freq (4.3.8,
+// Table 17) The Ethernet controller is in D2 domain and runs at hclk frequency
+enum {
+  D1CPRE = 1,  // actual divisor value
+  HPRE = 2,    // actual divisor value
+  D1PPRE = 4,  // register values, divisor value = BIT(value - 3) = / 2
+  D2PPRE1 = 4,
+  D2PPRE2 = 4,
+  D3PPRE = 4
+};
+// PLL1_P: odd division factors are not allowed (8.7.13) (according to Cube, '1'
+// is also an "odd division factor").
+// Make sure your chip is revision 'V', otherwise set PLL1_N = 400
+enum { PLL1_HSI = 64, PLL1_M = 32, PLL1_N = 480, PLL1_P = 2 };
+#define FLASH_LATENCY 0x24  // WRHIGHFREQ LATENCY
+#define SYS_FREQUENCY ((PLL1_HSI * PLL1_N / PLL1_M / PLL1_P / D1CPRE) * 1000000)
```

Comment:
There is no "sys" in H7, that is why I used "cpu".
There are three domains each one with its AHB clock and the CPU has a different clock. There is also a prescaler between the source (which is called "sys" in the datasheets) and the cpu, so using "sys" is misleading.

Reply:
Got it - reverted to CPU_FREQUENCY